src/OrchardCore.Modules/OrchardCore.Redis/Services/RedisBus.cs

```diff
@@ -33,6 +33,12 @@ public async Task SubscribeAsync(string channel, Action<string, string> handler)
                 await _redis.ConnectAsync();
             }
 
+            if (_redis.Connection == null)
+            {
+                _logger.LogError("Unable to subscribe to the channel '{ChannelName}'.", _channelPrefix + channel);
+                return;
```

Comment:
By this I meant.. is the bus used in places where you might want to make a different decisions based off if you can actually connect or not?  If I'm not mistaken this is used for things like the distributed reloading of shells for instance?

Reply:
> If I'm not mistaken this is used for things like the distributed reloading of shells for instance?

`RedisBus` is there but currently we never use it, to keep in sync the distributed cache with the database we use identifiers in place of using message bus events to invalidate cache entries.

> This hides what used to be an explicit exception before. Do you really want that change in behavior?

If `Connection` is still null it means that we can't connect and that just before `_redis.ConnectAsync()` already has logged an explicit exception. Then, without this null check and if `Connection` is null, in the subsequent try catch block, on `Connection.GetSubscriber()` it would throw another non relevant NRE + stack trace.

So it works as before, it just prevents useless logging.

I did the same in `RedisLock` and `RedisTagCache` where we check `.Database` in place of `.Connection` but that's quite the same. Then, about using a try catch block (that was already there), for the redis lock, it still respects the contract, it still uses its retry logic and return false after the timeout.

The `RedisTagCache` is more related to dynamic caching (see the related comments), so I think it's okay to use a try catch block and just log errors, for the `RedisBus` I'm open to any suggestion but as said we opted to not use it to keep in sync data e.g. in a distributed env.